Data/DataModel/ContentFile.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text;
+
+namespace ServerCore.DataModel
+{
+    /// <summary>
+    /// Database tracking for puzzle files stored on the server
+    /// </summary>
+    public class ContentFile
+    {
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int ID { get; set; }
+
+        /// <summary>
+        /// Identifies how this content should be handled
+        /// </summary>
+        public ContentFileType FileType { get; set; }
+
+        /// <summary>
+        /// Required for indexing. Use <see cref="Event"/> instead.
+        /// </summary>
+        public int EventID { get; set; }
```

Comment:
Why do both EventID and Event properties appear? We do not do this anywhere else. Should we?

Reply:
We normally don't need to (EF will typically generate these properties on its own), but because I'm setting up an explicit index using the ID, it needs to be referenceable. EF recognizes the Foo and FooID pattern to mean that Foo is a navigation property backed by FooID.